src/porepy/models/constitutive_laws.py

```diff
@@ -1085,48 +1108,35 @@ def solid_thermal_conductivity(self, subdomains: list[pp.Grid]) -> pp.ad.Operato
         """
         return Scalar(self.solid.thermal_conductivity(), "solid_thermal_conductivity")
 
-    def thermal_conductivity(self, subdomains: list[pp.Grid]) -> np.ndarray:
+    def thermal_conductivity(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
         """Thermal conductivity [m^2].
 
         The thermal conductivity is computed as the porosity-weighted average of the
         fluid and solid thermal conductivities. In this implementation, both are
         considered constants, however, if the porosity changes with time, the weighting
         factor will also change.
 
-        The thermal conductivity is quantity which enters the discretized equations in a
-        form that cannot be differentiated by Ad (this is at least true for a subset of
-        the relevant discretizations). For this reason, the thermal conductivity is not
-        returned as an Ad operator, but as a numpy array, to be wrapped as a
-        SecondOrderTensor and passed as a discretization parameter.
-
         Parameters:
-            subdomain: Subdomain where the thermal conductivity is defined.
+            subdomains: List of subdomains where the thermal conductivity is defined.
 
         Returns:
-            Cell-wise conducivity tensor.
+            Cell-wise conducivity operator.
 
         """
         assert len(subdomains) == 1, "Only one subdomain is allowed."
```

Comment:
Is not this an inconsistency compared to the permeability?

Reply:
Yes, fixed